src/puppetlabs/puppetdb/command.clj

```diff
@@ -131,27 +132,29 @@
           (string? mq-endpoint)
           (string? raw-command)]
    :post [(string? %)]}
+  ;; FIXME: is this efficient enough?  It's rebuilding the whole MQ
+  ;; stack every call.
   (let [id (kitchensink/uuid)]
-    (with-open [conn (mq/activemq-connection mq-spec)]
-      (mq/connect-and-publish! conn mq-endpoint raw-command {"received" (kitchensink/timestamp)
-                                                             "id" id}))
+    (with-open [factory (mq/activemq-connection-factory mq-spec)]
+      (mq/connect-and-publish! factory mq-endpoint raw-command
+                               {"received" (kitchensink/timestamp)
+                                "id" id}))
     id))
 
-(defn enqueue-command!
+(defn-validated enqueue-command! :- s/Str
```

Comment:
Question: when do we use defn-validated in this codebase? What's the motivation for using it here? Don't read too much in to this, I just want to know.